lib/queue.js

```diff
@@ -1,15 +1,31 @@
 class Queue {
+    static maxLenth = 20;
+    
     constructor() {
-        // this.store = ...
-        throw new Error("This method has not been implemented!");
+        this.store = new Array();
+        this.head = 0;
+        this.tail = 0;
     }
 
-    enqueue(element) {
-        throw new Error("This method has not been implemented!");
+    enqueue(value) {
+        let len = this.store.length;
+        if (this.head === this.tail && !this.isEmpty()) {
+            throw new Error("Queue is Full!")
+        }
+        this.store[len] = value;
+        this.tail = (this.tail + 1) % Queue.maxLenth;
     }
 
     dequeue() {
```

Comment:
You should also check to see if the `head` and `tail` are equal, if so the Queue is empty and you should do something to avoid removing `null` entries.